### DISCUSSION TITLE
Integrate a serial transport method with the FastDDS (the version 2.6.7 supported in ROS2 Humble)

### DISCUSSION THREAD

**Discussion by Syamkamalnadhc** [eProsima/Fast-DDS#4551] [Q&A]
We are using DDS as part of ROS2. Currently the transport protocol supported by Fast DDS are UDP, TCP and SHM.  Could you please suggest if the interfaces for serial communication protocols such as USB , PCIE are available for Fast DDS? Or a reference framework to integrate the custom protocol available? Currently our understanding is if we have to integrate a custom protocol, we have to develop the classes for Transport interface, Transport descriptor, SenderResource, Channel Resource etc and also should edit the XMLParser class.

**Comment by EduPonz** [comment]
Hi @Syamkamalnadhc,

At the moment there is no support for transports other than the ones you mentioned. You analysis of the task is correct. I may point you to our support team (support@eprosima.com) in case you are interested.

**Comment by Syamkamalnadhc** [comment]
We are trying to set the transport to a custom transport which is PCIe using xml profile. We have achieved the transmission of data using PCIe, but this was achieved by hardcoding the TransportDescrptor as PCIe in one of the HelloWorldExamples.
But while trying to switch the transport using an xml profile shared memory transport is being instantiated. 
 
Below is the loaded xml profile
 
<?xml version="1.0" encoding="utf-8" ?>
<profiles xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">
<transport_descriptors>
<transport_descriptor>
<transport_id>pcie_transport</transport_id>
<type>PCIe</type>
<device_id>10</device_id>
<read_buffer_size>20</read_buffer_size>
<write_buffer_size>30</write_buffer_size>
</transport_descriptor>
</transport_descriptors>
	<participant profile_name="DisableBuiltinTransportsParticipant">
<rtps>
<userTransports>
<transport_id>pcie_transport</transport_id>
</userTransports>
<useBuiltinTransports>false</useBuiltinTransports>
</rtps>
</participant>
</profiles>
